validator/db/kv/proposer_settings.go

```diff
@@ -0,0 +1,132 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	validatorServiceConfig "github.com/prysmaticlabs/prysm/v4/config/validator/service"
+	validatorpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1/validator-client"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+	"google.golang.org/protobuf/proto"
+)
+
+// NoProposerSettingsFound is an error thrown when no settings are found in bucket
+var NoProposerSettingsFound = errors.New("no proposer settings found in bucket")
+
+// UpdateProposerSettingsForPubkey updates the existing settings for an internal representation of the proposers settings file at a particular public key
+func (s *Store) UpdateProposerSettingsForPubkey(ctx context.Context, pubkey [fieldparams.BLSPubkeyLength]byte, options *validatorServiceConfig.ProposerOption) error {
+	_, span := trace.StartSpan(ctx, "validator.db.UpdateProposerSettingsDefault")
+	defer span.End()
+	err := s.db.Update(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(proposerSettingsBucket)
+		b := bkt.Get(proposerSettingsKey)
+		if len(b) == 0 {
+			return fmt.Errorf("no proposer settings found in bucket")
+		}
+		to := &validatorpb.ProposerSettingsPayload{}
+		if err := proto.Unmarshal(b, to); err != nil {
+			return errors.Wrap(err, "failed to unmarshal proposer settings")
+		}
+		settings, err := validatorServiceConfig.ToSettings(to)
```

Comment:
why can't we just modify the proto file directly here ? Or is it just much simpler to update `settings` ?

Reply:
I can change this, it's just options was passed as the internal struct representation so i need to convert it into a proto type anyways. I can change this to converting the options to a proto type to update instead or just update the values of the proto if it's preferred.